code/modules/projectiles/boxes_magazines/generic_ammo_box.dm

```diff
@@ -0,0 +1,63 @@
+/obj/item/ammo_box/generic
+	name = "generic ammo box"
+	desc = "A generic, unbranded box of ammo. It doesn't have great capacity, but it can hold a variety of different calibers."
+	max_ammo = 20
+	start_empty = TRUE
+	icon_state = "generic-ammo"
+	/// Does the box currently have an ammo type set?
+	var/ammo_set = FALSE
+	/// Name of the currently set ammo type
+	var/ammo_name
+
+/obj/item/ammo_box/generic/update_ammo_count()
+	. = ..()
+	if(LAZYLEN(stored_ammo) == 0)
+		ammo_set = FALSE
+		ammo_type = /obj/item/ammo_casing
+
+/obj/item/ammo_box/generic/proc/update_max_ammo(obj/item/ammo_casing/ammo)
```

Comment:
I really don't like this proc at all, i'll think about a better method

Reply:
Agreeded. you can just fetch a bullets_per_box from the ammo_casing.

Reply:
took falcon's suggestion, update_max_ammo is now handled by a bullet_per_box variable rather than a list checking calibers